backend/remote-state/azure/backend.go

```diff
@@ -51,6 +51,13 @@ func New() backend.Backend {
 				DefaultFunc: schema.EnvDefaultFunc("ARM_SAS_TOKEN", ""),
 			},
 
+			"versioning": {
```

Comment:
whilst the S3 Backend calls this "versioning" (due to that being [the name of the feature in S3](https://docs.aws.amazon.com/AmazonS3/latest/user-guide/enable-versioning.html)) - since Azure's calling this "Snapshots" instead could we use that terminology instead?

```suggestion
			"snapshot": {
```